src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -1,25 +1,121 @@
-    private final EiffelPluginConfiguration pluginConfig;
+    private String eiffelMessage;
+    private String eiffelType;
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
 
     public EiffelEventSender(EiffelPluginConfiguration pluginConfig) {
+        this.httpRequest = new HttpRequest(HttpMethod.POST);
         this.pluginConfig = pluginConfig;
     }
 
-    public void send(EiffelEvent eiffelEvent) {
-        JsonObject eiffelEventJson = gson.toJsonTree(eiffelEvent).getAsJsonObject();
-        System.out.println("Event to send::: " + eiffelEventJson.toString());
-        System.out.println("REMReM URL ::: " + pluginConfig.getRemremPublishURL());
-        // TODO Implement event sending using REMReM details from pluginconfig.
+    public EiffelEventSender(EiffelPluginConfiguration pluginConfig, HttpRequest httpRequest) {
+        this.httpRequest = httpRequest;
+        this.pluginConfig = pluginConfig;
+    }
+
+    /**
+     * Sends a REMReM eiffel message using
+     *
+     */
+    public void send() {
+        try {
+            generateAndPublish();
+        } catch (URISyntaxException | IOException | EiffelEventSenderException e) {
+            LOGGER.error("Failed to send eiffel message.", e);
+        }
+    }
+
+    public EiffelEventSender setMessage(final EiffelEvent eiffelMessage) {
+        this.eiffelMessage = gson.toJsonTree(eiffelMessage).toString();
+        return this;
+    }
+
+    public EiffelEventSender setType(final String eiffelType) {
+        this.eiffelType = eiffelType;
+        return this;
     }
 
+    private void generateAndPublish()
+            throws URISyntaxException, IOException, EiffelEventSenderException {
+        if (!isConfigurationSet()) {
+            final String errorMessage = String.format(
+                    "Neccessary configuration is missing to send event."
+                            + "\neiffelMessage: %s\neiffelType: %s\neiffelProtocol: %s",
+                    eiffelMessage, eiffelType, EIFFEL_PROTOCOL);
+            throw new EiffelEventSenderException(errorMessage);
+        }
+
+        httpRequest = assembleRequest(httpRequest);
+        final ResponseEntity response = httpRequest.performRequest();
+
+        final int statusCode = response.getStatusCode();
+        final String result = response.getBody();
+
+        if (HttpStatus.SC_OK == statusCode) {
+            LOGGER.info("Generated and published eiffel message successfully. \n{}", result);
+        } else {
+            final String errorMessage = String.format(
+                    "Could not generate and publish eiffel message due to server issue or invalid json data, "
+                            + "Status Code :: %d\npublishURL :: %s\ninput message :: %s\nError Message  :: %s",
+                    statusCode, httpRequest.getBaseUrl(), eiffelMessage, result);
+            throw new EiffelEventSenderException(errorMessage);
+        }
```

Comment:
You could break this out to its own method

Reply:
So send HttpResponse to a verification method?

Reply:
This is now placed in verifyResponse method.